modules/customization/proc-customize-rhdh-sidebar-menuitems.adoc

```diff
@@ -0,0 +1,116 @@
+
+You can customize the sidebar menu items in your `app-config-rhdh.yaml` file. The order and parent-child relationships of sidebar main menu items can be customized using the `dynamicPlugins.frontend.default.main-menu-items.menuItems` field for the main menu items, or the `dynamicPlugins.frontend.<package_name>.menuItems` field for dynamic plugin menu items in the `app-config-rhdh.yaml` file:
+
+[source,yaml]
+----
+dynamicPlugins:
+  frontend:
+    <package_name>: # same as `scalprum.name` key in plugin's `package.json`
+      menuItems: # optional, allows you to configure plugin menu items in the main sidebar navigation
+        <menu_item_name>: # unique name in the plugin menu items list <1>
+          icon: fooIcon # optional, same as `menuItem.icon` in `dynamicRoutes` <2>
+          title: Foo Plugin Page # optional, same as `menuItem.text` in `dynamicRoutes` <3>
+          priority: 10 # optional, defines the order of menu items in the sidebar <4>
+          parent: favorites # optional, defines parent-child relationships for nested menu items <5>
+----
+
+<1> This attribute represents a unique name in the main sidebar navigation. It can denote either a standalone menu item or a parent menu item. If this attribute represents a plugin menu item, the name of the attribute must match with the corresponding path in `dynamicRoutes`. For example, if `dynamicRoutes` defines `path: /my-plugin`, then `menu_item_name` must be defined as `my-plugin`.
++
+--
+For more complex, multi-segment paths such as `path: /metrics/users/info`, the `menu_item_name` must use dot notation to represent the full path, for example, `metrics.users.info`. Trailing and leading slashes in paths are ignored. For example, `path: /docs` results in `menu_item_name: docs`, and `path: /metrics/users` results in `menu_item_name: metrics.users`.
+--
+
+<2> This optional attribute specifies the icon for the menu item. You can use default icons or extend the icon set with dynamic plugins. {product-short} also provides additional icons in its internal library, such as: 
++
+--
+.Icons in the internal library =
+[source]
+----
+  home: from '@mui/icons-material/HomeOutlined'
+  group: from '@mui/icons-material/People'
+  category: from '@mui/icons-material/CategoryOutlined'
+  extension: from '@mui/icons-material/ExtensionOutlined'
+  school: from '@mui/icons-material/SchoolOutlined'
+----
+
+If the icon is already defined in the `dynamicRoutes` configuration under `menuItem.icon`, it can be removed from the in the `menuItems` configuration. Also, both SVG and HTML image icons are supported. For example:
+
+.Example SVG icon
+[source,html]
+----
+icon: <svg width="20px" height="20px" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg" fill="#ffffff">...</svg>
+----
```

Comment:
Consider using a more consistent approach, where all example follow the same pattern, and the example is a YAML snippet, such as:

Internal Library icons::
--

.Home
[source,yaml]
----
dynamicPlugins:
  frontend:
    <package_name>:
      menuItems:
        <menu_item_name>: 
          icon: home
----

.Group
[source,yaml]
----
dynamicPlugins:
  frontend:
    <package_name>:
      menuItems:
        <menu_item_name>: 
          icon: group
----

.Category
[source,yaml]
----
dynamicPlugins:
  frontend:
    <package_name>:
      menuItems:
        <menu_item_name>: 
          icon: category
----

.Extension
[source,yaml]
----
dynamicPlugins:
  frontend:
    <package_name>:
      menuItems:
        <menu_item_name>: 
          icon: extension
----

.School
[source,yaml]
----
dynamicPlugins:
  frontend:
    <package_name>:
      menuItems:
        <menu_item_name>: 
          icon: school
----
--

SVG icon::
--
Use an inline SVG image.

[source,yaml]
----
dynamicPlugins:
  frontend:
    <package_name>:
      menuItems:
        <menu_item_name>: 
          icon: <svg width="20px" height="20px" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg" fill="#ffffff">...</svg>
----

Image icon::
--
Use an external image available as an HTTPS resource.

[source,yaml]
----
dynamicPlugins:
  frontend:
    <package_name>:
      menuItems:
        <menu_item_name>: 
          icon: https://img.icons8.com/ios-glyphs/20/FFFFFF/shop.png
----
--

--